### ISSUE TITLE
Support domainname for runtime spec

### ISSUE THREAD

**Issue by liubin** [kata-containers/kata-containers#5960]
Refer to:

- https://github.com/opencontainers/runtime-spec/pull/1156
- https://github.com/opencontainers/runc/pull/3600

This need to add to both runtime and runtime-rs.

**Comment by dcantah** [comment]
Can assign me here

**Comment by dcantah** [comment]
> This need to add to both runtime and runtime-rs.

This shouldn't have a runtime component, right? I'd imagine it's just work in the agent (rustjail specifically).

**Comment by dcantah** [comment]
Ahh, I guess we'd need the updated spec

**Comment by liubin** [comment]
> > This need to add to both runtime and runtime-rs.
> 
> This shouldn't have a runtime component, right? I'd imagine it's just work in the agent (rustjail specifically).

Yes, if there is no need to update the `domainame` on the Golang runtime side, maybe we need only add the property to the agent protocol.

But for runtime-rs, the [spec](https://github.com/kata-containers/kata-containers/tree/main/src/libs/oci/src) should add this new field. You can implement runtime/runtime-rs in two PRs.

BTW, the agent OCI protocol is [here](https://github.com/kata-containers/kata-containers/blob/main/src/libs/protocols/protos/oci.proto)
